### PR TITLE
[SPARK-25202][SQL][FOLLOW-UP] Keep the old parameter name 'pattern' at split in Scala API

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

To address the concern pointed out in https://github.com/apache/spark/pull/22227. This will make `split` source-compatible by removing minimal cosmetic changes.

### Why are the changes needed?

For source compatibility.

### Does this PR introduce any user-facing change?

No (it will prevent potential user-facing change from the original PR)

### How was this patch tested?

Unittest was changed (in order for us to detect that source compatibility easily).

### CODE REVIEW

**Review by cloud-fan — APPROVED** [review]

**Comment by maropu** [comment]
retest this please

**Review by maropu — APPROVED** [review]

**Review by srowen — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -2463,22 +2463,22 @@ object functions {
    * Splits str around matches of the given regex.
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: nit: This general comment refers to parameters, it seems. And `pattern` was replaced by `regex` here in in the original PR. How about _Splits str around matches of the given **pattern**_

**Review by dongjoon-hyun — APPROVED** [review]

**Review by viirya — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Oops, mistake. I was removing my old branches.

**Comment by HyukjinKwon** [comment]
Thanks guys. I am going to merge this because the last change was comment-only. I locally ran the Scala linter to make sure.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.0.

**Pull request closed without merging**